suppression_list.go

```diff
@@ -174,36 +217,54 @@ func (c *Client) SuppressionUpsertContext(ctx context.Context, entries []Suppres
 }
 
 // Wraps call to server and unmarshals response
-func (c *Client) suppressionGet(ctx context.Context, finalURL string) (*SuppressionListWrapper, *Response, error) {
+func (c *Client) suppressionGet(ctx context.Context, finalURL string, sp *SuppressionPage) (*Response, error) {
+
 	// Send off our request
 	res, err := c.HttpGet(ctx, finalURL)
 	if err != nil {
-		return nil, res, err
+		return res, err
 	}
 
 	// Assert that we got a JSON Content-Type back
 	if err = res.AssertJson(); err != nil {
-		return nil, res, err
+		return res, err
 	}
 
 	err = res.ParseResponse()
 	if err != nil {
-		return nil, res, err
+		return res, err
 	}
 
 	// Get the Content
 	bodyBytes, err := res.ReadBody()
 	if err != nil {
-		return nil, res, err
+		return res, err
 	}
 
 	// Parse expected response structure
-	var resMap SuppressionListWrapper
-	err = json.Unmarshal(bodyBytes, &resMap)
+	err = json.Unmarshal(bodyBytes, sp)
+
+	// For usage convenience parse out common links
+	for _, link := range sp.Links {
+		switch link.Rel {
+		case "next":
+			sp.NextPage = link.Href
+		case "previous":
+			sp.PrevPage = link.Href
+		case "first":
+			sp.FirstPage = link.Href
+		case "last":
+			sp.LastPage = link.Href
+		}
+	}
+
+	if sp.client == nil {
+		sp.client = c
+	}
 
 	if err != nil {
-		return nil, res, err
+		return res, err
```

Comment:
Please move this error block up next to the `Unmarshal` and change the last line in the function to `return res, nil`